bgpd/bgp_updgrp_adv.c

```diff
@@ -480,6 +481,23 @@ void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
 			return;
 	}
 
+	/* Check if we are sending the same route. This is needed to
+	 * avoid duplicate UPDATES. For instance, filtering communities
+	 * at egress, neighbors will see duplicate UPDATES despite
+	 * the route wasn't changed actually.
+	 */
+	if (adj->attr_hash == attr_hash) {
```

Comment:
I think I would I would want the test to be `if (adj->attr_hash && adj->attr_hash == attr_hash)` This way we stop the side case where the first send to the peer happens to not be sent because the attrhash_key_make returns 0